vendor/golang.org/x/sys/unix/zsyscall_openbsd_riscv64.go

```diff
@@ -527,6 +527,14 @@ func ioctl(fd int, req uint, arg uintptr) (err error) {
 	return
 }
 
+func ioctlPtr(fd int, req uint, arg unsafe.Pointer) (err error) {
+	_, _, e1 := syscall_syscall(libc_ioctl_trampoline_addr, uintptr(fd), uintptr(req), uintptr(arg))
+	if e1 != 0 {
+		err = errnoErr(e1)
+	}
+	return
+}
+
 var libc_ioctl_trampoline_addr uintptr
 
 //go:cgo_import_dynamic libc_ioctl ioctl "libc.so"
```

Comment:
It seems like the code patch is adding a new function `ioctlPtr` that takes an unsafe pointer as the third argument. This function uses `syscall_syscall` to call `libc_ioctl_trampoline_addr` with the given arguments and returns any error encountered during the system call. 

One potential improvement would be to add more context or documentation on when and where this function should be used, as well as any potential edge cases or related functions that could affect its behavior. Additionally, since `arg` is an unsafe pointer, there may be a risk of data corruption or security vulnerabilities if it is not properly checked or validated before being passed to the syscall. It may be worth considering using a safer type, such as a slice or struct, instead of an unsafe pointer.